db.go

```diff
@@ -597,16 +597,40 @@ func (m *DbMap) Select(ctx context.Context, i interface{}, query string, args ..
 		expandSliceArgs(&query, args...)
 	}
 
+	args, err := m.convertArgs(args...)
+	if err != nil {
+		return nil, err
+	}
+
 	return hookedselect(ctx, m, m, i, query, args...)
 }
 
+func (m *DbMap) convertArgs(args ...interface{}) ([]interface{}, error) {
```

Comment:
nit: rather than sandwiching this between Select and ExecContext, move this towards the bottom of the file next to `expandSliceArgs`, which is also used as a helper by all of these functions.

Reply:
Good call, done.